### ISSUE TITLE
XML matched incorrectly

### ISSUE THREAD

**Issue by AJenbo** [eranif/codelite#1646]
CL will highlight <CASEID> as the match for </CASE> with the following XML-file:
```
<CASE>
    <CASEID></CASEID>
</CASE>
```
Note that this issue only occure when there is a nested tag that starts with the same string as the actuall tag. The following cases works fine:

```
<CASEID></CASEID>
<CASE>
    <CASE></CASE>
    <IDCASE></IDCASE>
</CASE>
<CASEID></CASEID>
```

Original report:
https://forums.codelite.org/viewtopic.php?f=13&t=3660

**Comment by w4du** [comment]
I think it should be reopened since it's not fixed as I tested in version 11.0.6(Win7 x64)

**Comment by pattisahusiwa** [comment]
Agreed. Working on this sample,
````xml
<CASEID>
    <IDCASE>
        <CASEID>
            <CASE>
                <CASE>Test</CASE>
                <IDCASE>Test</IDCASE>
            </CASE>
        </CASEID>
    </IDCASE>
</CASEID>
````
but not this one,
````xml
<catalog>
    <catalog_item>Test</catalog_item>
</catalog>
````
